File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/DayValueFormatter.java
package io.github.zwieback.familyfinance.business.chart.service.formatter;

import com.github.mikephil.charting.formatter.ValueFormatter;

import org.threeten.bp.LocalDate;

import io.github.zwieback.familyfinance.util.DateUtils;

public class DayValueFormatter extends ValueFormatter {

    @Override
    public String getFormattedValue(float daysFromEpoch) {
        LocalDate day = determineCorrectDay(daysFromEpoch);
        return DateUtils.localDateToString(day);
    }

    private static LocalDate determineCorrectDay(float daysFromEpoch) {
        return DateUtils.epochDayToLocalDate((long) daysFromEpoch);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/DayValueFormatter.kt
package io.github.zwieback.familyfinance.business.chart.service.formatter

import io.github.zwieback.familyfinance.extension.DaysFromEpoch
import org.threeten.bp.format.DateTimeFormatter

class DayValueFormatter : SpecificPeriodFormatter() {

    override fun getFormatter(): DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_DATE

    override fun getSpecificPeriod(value: Long) = DaysFromEpoch(value)
}
